File: kunal/1291. Sequential Digits.java
package kunal;

import java.util.ArrayList;
import java.util.List;

class Solution {
    public List<Integer> sequentialDigits(int low, int high) {
        String fixed = "123456789";

        List<Integer> ans = new ArrayList<>();

        for (int digit = String.valueOf(low).length(); digit <= String.valueOf(high).length(); digit++) {
            int left = 0;
            int right = left + digit;

            while (right <= fixed.length()) {
                String subString = fixed.substring(left, right);
                if (Integer.parseInt(subString) >= low && Integer.parseInt(subString) <= high) {
                    ans.add(Integer.parseInt(subString));
                }
                left++;
                right++;
            }
        }

        return ans;
    }
}

File: kunal/1291. Sequential Digits.py
class Solution:
    def sequentialDigits(self, low: int, high: int) -> list[int]:
        fixed = "123456789"

        ans = []

        for i in range(len(str(low)), len(str(high)) + 1):
            left = 0
            right = left + i

            while right <= len(fixed):
                new_int = int(fixed[left:right])
                ans.append(new_int) if new_int >= low and new_int <= high else None
                left += 1
                right += 1

        return ans


print(Solution().sequentialDigits(100, 300))
print(Solution().sequentialDigits(1000, 13000))
